torch/csrc/autograd/python_variable.cpp

```diff
@@ -282,7 +292,7 @@ int THPVariable_set_data(THPVariable *self, PyObject *data)
   if (&self->cdata.data().type() != &tensor.type()) {
     // we change the type of var.data so we must change the type of var
     auto newType = VariableType::getType(tensor);
-    self->cdata.get()->*get(TensorImpl_Type()) = newType;
```

Comment:
There are no more call sites of `TensorImpl_Type`. This means you should delete the hack, right?

```
// XXX: This is a hack to access private TensorImpl::type_
// http://bloglitb.blogspot.com/2011/12/access-to-private-members-safer.html
// This is currently needed because module.float() changes the type of the
// data field of each variable. We should fix this and not allow changing the
// type of var.data.
```

But this is not the end of the story. See `set_type` for more.